onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -794,9 +797,51 @@ static Status GetValidatedEpContextPath(const std::filesystem::path& ep_context_
   return Status::OK();
 }
 
+// Serialize an EPContext model into a onnx::ModelProto.
+static Status EpContextModelToProto(const onnxruntime::Model& ep_context_model,
```

Comment:
Should this belong to either tensorprotoutils or EpContenxt. Paritioninaer sees out of place.

Reply:
Yeah, I agree. Unfortunately, all of the EPContext generation code is currently in the partitioner at the moment. I did the following:
- moved this specific function to a separate file (ep_context_utils.h/cc)
- Added TODO comments on other existing EPContext-related functions to note that they should also be moved out of the partitioner. I didn't do that move now because that would obfuscate the actual changes to those functions.